src/default-markup.ts

```diff
@@ -0,0 +1,235 @@
+        height: ${props.height || 'auto'};
+      `,
+      solidPlaceholder: `
+        background-color: ${props.backgroundColor ||
+          props.dominateColor ||
+          'inherit'};
+        width: ${props.width || 'auto'};
+        height: ${props.height || 'auto'};
+      `,
+      imagePlaceholder: `
+        width: ${props.width || 'auto'};
+        height: ${props.height || 'auto'};
+        object-fit: cover;
+        object-position: center center;
+        filter: blur(50px);
+      `,
+      svgPlaceholder: `
+        position: absolute;
+        top: 0; bottom: 0; left: 0; right: 0;
+        width: 100%;
+        height: 100%;
+        object-fit: cover;
+        object-position: center center;
+      `,
+      imageTag: `
+        position: absolute;
+        top: 0; bottom: 0; left: 0; right: 0;
+        width: 100%;
+        height: 100%;
+        object-fit: cover;
+        object-position: center center;
+      `,
+    },
+  }
+
+  if (props.sharpMethod === 'fluid') {
+    return `
+      <div class="gria-image-wrapper" style="${styles.fluid.imageWrapper}">
+
+        ${comment('preserve the aspect ratio')}
+        <div class="gria-image-padding" style="${
+          styles.fluid.imagePadding
+        }"></div>
+
+        ${comment('show a solid background color.')}
+        <div
+          class="gria-solid-placeholder"
+          title="${alt}"
+          style="${styles.fluid.solidPlaceholder}">
+        </div>
+
+        ${comment('show the blurry base64 image.')}
+        <img
+          class="gria-base64-placeholder"
+          src="${props.base64}"
+          title="${alt}"
+          alt="${alt}"
+          style="${styles.fluid.imagePlaceholder}"
+        >
+
+        ${comment('show a traced SVG image.')}
+        <img
+          class="gria-tracedSVG-placeholder"
+          src="${props.tracedSVG}"
+          title="${alt}"
+          alt="${alt}"
+          style="${styles.fluid.svgPlaceholder}"
+        >
+
+        ${comment('load the image sources.')}
+        <picture classname="gria-image-sources">
+          <source srcset="${props.srcSet}" sizes="${props.sizes}">
+          <img
+            src="${src}"
+            srcset="${props.srcSet}"
+            sizes="${props.sizes}"
+            title="${alt}"
+            alt="${alt}"
+            loading="lazy"
+            style="${styles.fluid.imageTag}"
+          >
+        </picture>
+
+        ${comment('(inefficiently) add noscript support.')}
+        <noscript>
+          <picture classname="gria-image-sources">
+            <source srcset="${props.srcSet}" sizes="${props.sizes}">
+            <img
+              src="${src}"
+              srcset="${props.srcSet}"
+              sizes="${props.sizes}"
+              title="${alt}"
+              alt="${alt}"
+              loading="lazy"
+              style="${styles.fluid.imageTag}"
+            >
+          </picture>
+        </noscript>
+      </div>
+    `
+  }
```

Comment:
ah I've just seen this, I wonder why do we need noscript support, since all the markup are already static and not relying on any script anyway?

Reply:
I'll remove this since it add duplicate images if noscript is enabled 😅 In the screenshot below, I add `left: 100px` css to the content of noscript so we can see both images

<img width="832" alt="image" src="https://user-images.githubusercontent.com/3383539/66059637-4a2f6600-e577-11e9-981d-6d970d69e946.png">